dotnet/src/Connectors/Connectors.AssemblyAI/AudioToText/AssemblyAIAudioToTextService.cs

```diff
@@ -0,0 +1,282 @@
+        var url = this.Url("v2/transcript");
+        var jsonRequest = new JsonObject();
+        jsonRequest["audio_url"] = audioUrl;
+        if (executionSettings?.ExtensionData is not null)
+        {
+            foreach (var attribute in executionSettings.ExtensionData)
+            {
+                jsonRequest[attribute.Key] = JsonValue.Create(attribute.Value);
+            }
+        }
+
+        using var request = HttpRequest.CreatePostRequest(url, jsonRequest);
+        this.AddDefaultHeaders(request);
+        using var response = await this.SendWithSuccessCheckAsync(this._httpClient, request, ct).ConfigureAwait(false);
+        var jsonStream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false);
+        var json = await JsonDocument.ParseAsync(jsonStream, cancellationToken: ct).ConfigureAwait(false);
+        if (json.RootElement.TryGetProperty("error", out var property))
+        {
+            throw new KernelException($"Failed to create transcript. Reason: {property.GetString()!}");
+        }
+
+        return json.RootElement.GetProperty("id").GetString()!;
+    }
+
+    private async Task<JsonDocument> WaitForTranscriptToProcessAsync(
+        string transcriptId,
+        PromptExecutionSettings? executionSettings = null,
+        CancellationToken ct = default
+    )
+    {
+        var url = this.Url($"v2/transcript/{transcriptId}");
+        var pollingInterval = TimeSpan.FromSeconds(1);
+        if (executionSettings is AssemblyAIAudioToTextExecutionSettings aaiSettings)
+        {
+            pollingInterval = aaiSettings.PollingInterval;
+        }
+
+        while (!ct.IsCancellationRequested)
+        {
+            using var request = HttpRequest.CreateGetRequest(url);
+            this.AddDefaultHeaders(request);
+            using var response = await this.SendWithSuccessCheckAsync(this._httpClient, request, ct).ConfigureAwait(false);
+            var jsonStream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false);
+            var json = await JsonDocument.ParseAsync(jsonStream, cancellationToken: ct).ConfigureAwait(false);
+
+            var status = json.RootElement.GetProperty("status").GetString()!;
+            switch (status)
+            {
+                case "processing":
+                case "queued":
+                    await Task.Delay(pollingInterval, ct).ConfigureAwait(false);
+                    break;
+                case "completed":
+                    return json;
+                case "error":
+                    var errorString = json.RootElement.GetProperty("error").GetString()!;
+                    throw new KernelException($"Failed to create transcript. Reason: {errorString}");
+                default:
+                    throw new KernelException("Unexpected transcript status. This code shouldn't be reachable.");
+            }
+        }
+
+        ct.ThrowIfCancellationRequested();
+        throw new KernelException("This code is unreachable.");
```

Comment:
If the code is unreachable, what is it for?

Reply:
If `ct.ThrowIfCancellationRequested()` is reached, then the cancellation token is cancelled, so `ct.ThrowIfCancellationRequested()` will throw.
Thus the next line is unreachable, but for compiler reasons, the method needs to return something or throw.